filebeat/harvester/log.go

```diff
@@ -30,15 +35,15 @@ func (h *Harvester) Harvest() {
 	//       for new lines in input stream. Simple 8-bit based encodings, or plain
 	//       don't require 'complicated' logic.
 	cfg := h.Config
-	readerConfig := logFileReaderConfig{
-		forceClose:         cfg.ForceCloseFiles,
-		closeOlder:         cfg.CloseOlder,
-		backoffDuration:    cfg.Backoff,
-		maxBackoffDuration: cfg.MaxBackoff,
-		backoffFactor:      cfg.BackoffFactor,
+	readerConfig := reader.LogFileReaderConfig{
+		ForceClose:         cfg.ForceCloseFiles,
+		CloseOlder:         cfg.CloseOlder,
+		BackoffDuration:    cfg.Backoff,
+		MaxBackoffDuration: cfg.MaxBackoff,
+		BackoffFactor:      cfg.BackoffFactor,
 	}
 
-	reader, err := createLineReader(
+	rd, err := reader.CreateLineReader(
```

Comment:
the reader pipeline being build via `CreateLine` is custom to log based harvester. Same goes for `LogFileReaderConfig`. For turning prospectors/harvesters into more modular form, every harvester should setup it's own reader pipeline. Plus have potentially custom file state (for supporting other types then tail). That is, I'd prefer some `havester/logs` package with `LogFileReaderConfig` and `CreateLineReader` being  private to `harvester/log` package.